python/ray/serve/_private/router.py

```diff
@@ -524,6 +525,22 @@ async def schedule_and_send_request(
                     ray.cancel(obj_ref_gen)
 
                 raise
+            except ActorDiedError:
+                # Replica has died but controller hasn't notified the router yet.
+                # Don't consider this replica for requests in the future.
+                self._replica_scheduler.drop_replica(replica.replica_id)
+                logger.warning(
+                    f"{replica.replica_id} will not be considered for future "
+                    "requests because it has died."
+                )
+                raise
+            except ActorUnavailableError:
+                # There are network issues, or replica has died but GCS is down so
+                # ActorUnavailableError will be raised until GCS recovers. For the
+                # time being, invalidate the cache entry so that we don't try to
+                # send requests to this replica without actively probing.
+                self._replica_scheduler.invalidate_cache_entry(replica.replica_id)
```

Comment:
ditto to my other comment, suggest to make this `self._replica_scheduler.on_replica_actor_unavailable`